pkg/build/strategies/sti/sti.go

```diff
@@ -287,7 +284,7 @@ func (b *STI) Exists(config *api.Config) bool {
 		}
 	}
 
-	return previousImageExists && b.installedScripts[api.SaveArtifacts]
+	return previousImageExists
```

Comment:
I realized recently that this check (installedScripts) appears to always return true because we assume the image may contain a save-artifacts script, which we can't confirm without running the image... but i'm still uncomfortable with removing the check entirely in case that changes in the future.

i'd be ok with moving the check to its own method (eg hasSaveArtifacts()) but i'd rather not remove it entirely.